src/tests/test_utils/common_test_utils/src/ov_tensor_utils.cpp

```diff
@@ -314,141 +314,105 @@ inline double less_or_equal(double a, double b) {
     if (std::isnan(a) || std::isnan(b)) {
         res = false;
     } else if (std::isinf(b) && b > 0) {
-        // b is grater than any number or eq the +Inf
+        // b is greater than any number or eq the +Inf
         res = true;
     } else if (std::isinf(a) && a > 0) {
         res = false;
     } else {
-        res = (std::fabs(b - a) <= (std::fmax(std::fabs(a), std::fabs(b)) * eps) || a < b);
+        res = std::fabs(b - a) <= eps || a <= b;
     }
-    double eq_midle_res = std::fabs(b - a);
-    bool eq_res = (std::fabs(b - a) <= (std::fmax(std::fabs(a), std::fabs(b)) * eps));
     return res;
 }
 
-struct Error {
-    double max = 0.;
-    double mean = 0.;
-    size_t max_coordinate = 0;
-    size_t count = 0;
-    double threshold;
+template <typename T>
+inline bool check_value(const T& value) {
+    if (std::isnan(value) || std::isinf(value) || value >= std::numeric_limits<T>::max() ||
+        value <= std::numeric_limits<T>::min()) {
+        return false;
+    }
+    return true;
+}
+
+class Error {
+protected:
+    struct IncorrectValue {
+        size_t coordinate;
+        double actual_value, expected_value, abs_threshold, rel_threshold;
+
+        IncorrectValue(double in_actual_value,
+                       double in_expected_value,
+                       double in_abs_threshold,
+                       double in_rel_threshold,
+                       size_t in_coordinate)
+            : actual_value(in_actual_value),
+              expected_value(in_expected_value),
+              abs_threshold(in_abs_threshold),
+              rel_threshold(in_rel_threshold),
+              coordinate(in_coordinate) {}
+    };
 
-    Error(double _threshold) : threshold(_threshold) {}
+    std::vector<IncorrectValue> incorrect_values;
+    double abs_threshold, rel_threshold;
 
-    void update(double val, size_t i) {
-        if (less(max, val)) {
-            max = val;
-            max_coordinate = i;
+public:
+    Error(const double in_abs_threshold, const double in_rel_threshold)
+        : abs_threshold(in_abs_threshold),
+          rel_threshold(in_rel_threshold) {}
+
+    void update(double actual, double expected, size_t coordinate) {
+        const auto diff = std::fabs(expected - actual);
+
+        const auto calculated_abs_threshold = abs_threshold * expected;
+        const auto calculated_rel_threshold = calculated_abs_threshold ? diff / calculated_abs_threshold : 1.;
+        if (less_or_equal(diff, calculated_abs_threshold) && less_or_equal(calculated_rel_threshold, rel_threshold)) {
+            return;
         }
-        mean += val;
-        count += less(threshold, val);
+        incorrect_values.emplace_back(
+            IncorrectValue(actual, expected, calculated_abs_threshold, calculated_rel_threshold, coordinate));
```

Comment:
```suggestion
        incorrect_values.emplace_back(actual, expected, calculated_abs_threshold, calculated_rel_threshold, coordinate);
```

`emplace_back` should create object inside by provided arguments